### DISCUSSION TITLE
Repository config: Import schema files on repository creation?

### DISCUSSION THREAD

**Discussion by erikgb** [eclipse-rdf4j/rdf4j#4440] [Q&A] [answered]
I am writing some tests for an application using RDF4J to access some GraphDB repositories. The tests do not strictly require GDB, and I would like to use a simple RDF4J LocalRepositoryManager to stub out GDB. The application under test creates some repositories, and I have hooks to replace the TTL file-templates, that are tailored for GDB config.

I wonder if RDF4J has something roughly equivalent to [graphdb:imports](https://graphdb.ontotext.com/documentation/10.0/configuring-a-repository.html#configuring-a-repository-configuration-parameters), or if I have to do this programmatically in my test-setup?

> A list of schema files that will be imported at startup. All statements found in these files will be loaded in the repository and will be treated as read-only.

**Comment by abrokenjester** [comment, marked as answer]
I don't think we have something comparable to that - so yeah, doing this during test setup is probably the easiest way forward.

**Reply by erikgb** [reply]
Thanks! Would it make sense to add a feature request for this? 😉

**Reply by abrokenjester** [reply]
I must admit I'm struggling a little to understand the utility of it: I mean what's the advantage of providing data files via a config property rather than just uploading them once via regular means?

However if the expected behaviour can be explained and it has value then I don't see why we _couldn't_ add something like that. Whether the exact same approach as GraphDB (a config property on the repo) is the right approach depends a bit on what we expect this to be used for.
